src/arrayMethodJoin.js

```diff
@@ -4,8 +4,28 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    const currentArrayLength = this.length;
+    const joinSeparator = separator + '';
+    let resultingString = '';
+
+    if (!currentArrayLength) {
+      return resultingString;
+    }
+
+    this.forEach((item, index) => {
+      /* const stringCharacter = item ?? '' is the best choice,
+      but I couldn`t disable eslint for the nullish operator */
+      const stringCharacter = item === null || item === undefined
```

Comment:
You can make it shorter using `==`
```js
console.log(undefined == null) // true

console.log(null == null)  // true

console.log(false == null) // false
```

Reply:
using == is very dangerous for our karma)